ckanext/s3filestore/uploader.py

```diff
@@ -632,10 +642,12 @@ def _get_resource_metadata(self):
         package = toolkit.get_action('package_show')(
             context={'ignore_auth': True}, data_dict={'id': self.resource['package_id']})
         metadata = {
-            'package_' + field: six.moves.urllib.parse.quote(package[field])
-            for field in package.keys() if isinstance(package[field], six.string_types)
+            'package_' + field: ensure_ascii(package[field])
+            for field in package.keys()
```

Comment:
Could brackets be added, this is semi confusing how it loops and drops item etc.

Reply:
Can't put brackets around the `for`, that's not valid syntax, but I've shuffled it around a bit to hopefully be more readable.